### PR TITLE
Improve cloneElement

### PR DESCRIPTION
**Pull request by emre1702**
Improvements:
1. Use virtual clone function for the elements (as wanted in TODO comment in CStaticFunctionDefinitions.cpp)
2. Created IsCloneable function for CElement
3. CObject copy constructor improved & fixed (add type "object", copy alpha, scale, health, freeze, doublesided, breakable, syncable, syncer)
4. Fixed IsSyncable() usage for CPed (used of new created CPed instead of the one element we want to clone)

### CODE REVIEW

File: Server/mods/deathmatch/logic/CPickup.cpp [hunk 1]
```diff
@@ -61,6 +61,23 @@ CPickup::~CPickup ( void )
 }
 
 
+CElement* CPickup::Clone ( bool* bAddEntity, CResource* pResource )
+{
+    CPickup* pTemp = m_pPickupManager->Create ( GetParentEntity () );
+    if ( pTemp )
+    {
+        pTemp->SetPickupType ( GetPickupType () );
+        pTemp->SetPickupType ( GetPickupType () );
```
Comment (xerox8521) [inline Server/mods/deathmatch/logic/CPickup.cpp]: Don't think setting the pickup type twice is required is it ?

Comment (emre1702) [reply Server/mods/deathmatch/logic/CPickup.cpp]: Removed, thx

**Comment by qaisjp** [comment]
Stop deleting your repository @emre1702!

**Comment by qaisjp** [comment]
I am working on this

**Comment by qaisjp** [comment]
This will auto close when `next` merges into `master`. Consider this merged.

**Pull request closed without merging**